runtime/src/rent_paying_accounts_by_partition.rs

```diff
@@ -45,7 +49,9 @@ impl RentPayingAccountsByPartition {
         &self,
         partition_end_index: PartitionIndex,
     ) -> &HashSet<Pubkey> {
-        &self.accounts[partition_end_index as usize]
+        self.accounts
+            .get(partition_end_index as usize)
+            .unwrap_or(&EMPTY_HASHSET)
```

Comment:
Wouldn't this just hide invalid lookup as opposed to addressing the root cause of invalid lookup index?

also, why not return an `Option` instead of using statics?

Reply:
Yes it will hide it.
This is not expected to happen on mnb where we are always at 432k slots per epoch.
This is almost certainly entirely a temporary issue with test or new clusters.
For those clusters, there should never be any rent paying accounts. So, returning empty is correct.

Reply:
the idea is that when we stop collecting rent, we will change these hashsets to vecs. And we will iterate them to identify the rent paying accounts in this partition index. RIght now we scan the accounts index by pubkey range. So, &Vec is ideal, &HashSet is ok for now, Option(&whatever) is ok, but could be confusing because it shouldn't ever fail.